frontend/components/Course/CourseSettings/CourseForm.tsx

```diff
@@ -138,6 +168,76 @@ const CourseForm = (props: CourseFormProps) => {
             });
     };
 
+    useEffect(() => {
+        const fetchData = async () => {
+            const loadedTags: Tag[] = await getTags(course.id);
+            const buildMap = {};
+
+            loadedTags.forEach((loadedTag) => {
+                buildMap[loadedTag.name] = loadedTag.id;
+            });
+            setOldTags(buildMap);
+
+            setTagLabels(
+                loadedTags.map((tag) => {
+                    return {
+                        label: tag.name,
+                        value: tag.name,
+                    };
+                })
+            );
+        };
+
+        fetchData();
+    }, [course]);
+
+    const handleCreateTag = async (inputValue: string) => {
+        if (!(inputValue in oldTags)) {
+            setAddedTags([...addedTags, inputValue]);
+        } else {
+            setDeletedTags(
+                deletedTags.filter((tag) => {
+                    return tag.name !== inputValue;
+                })
+            );
+        }
+
+        setTagLabels([...tagLabels, { label: inputValue, value: inputValue }]);
+    };
+
+    const handleTagChange = (_, event) => {
+        if (event.action === "remove-value") {
+            const text = event.removedValue.label;
+
+            if (text in oldTags) {
+                setDeletedTags([
+                    ...deletedTags,
+                    { name: text, id: oldTags[text] },
+                ]);
+            } else {
+                setAddedTags(
+                    addedTags.filter((tag) => {
+                        return tag !== text;
+                    })
+                );
+            }
+
+            setTagLabels(
+                tagLabels.filter((tagLabel) => {
+                    return tagLabel.label !== text;
+                })
+            );
+        } else if (event.action === "clear") {
+            setTagLabels([]);
+        } else if (event.action === "select-option") {
+            handleCreateTag(event.option.label);
+        }
+    };
+
+    useEffect(() => {
```

Comment:
This might not work as intended - consider the situation where some other input (say course name) was changed, and a tag was added and removed. But, the save button will still be disabled. This logic belongs to the isValid function.